code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java

```diff
@@ -456,127 +422,83 @@ protected void setUp() throws Exception
 	 * @param aPreString
 	 * @param expectedOutput
 	 */
-	private void PreTest(final String aPreString, final String expectedOutput)
+	@Test
+	void testPreWriter(final String aPreString, final String expectedOutput) throws PersistenceLayerException
 	{
-		Prerequisite prereq = null;
-		boolean bExceptionThrown = false;
-		boolean bExceptionExpected = expectedOutput.isEmpty();
-		try
+		for (int idx = 0; idx < (testparams.length >> 1); ++idx)
 		{
-			prereq = PreParserFactory.getInstance().parse(aPreString);
-		}
-		catch (PersistenceLayerException ple)
-		{
-			if (!bExceptionExpected)
+			Prerequisite prereq = null;
+			if (expectedOutput.isEmpty())
 			{
-				ple.printStackTrace();
-				fail("parse caused PersistenceLayerException: "
-					+ ple.toString());
+				assertThrows(PersistenceLayerException.class, () -> PreParserFactory.getInstance().parse(aPreString));
 			}
-			bExceptionThrown = true;
-		}
-
-		//
-		// Did we expect an exception to be thrown?
-		//
-		if (bExceptionExpected)
-		{
-			if (!bExceptionThrown)
+			else
 			{
-				if (prereq != null)
-				{
-					System.out.println(prereq.toString());
-				}
-				fail("exception expected but not thrown");
+				prereq = PreParserFactory.getInstance().parse(aPreString);
+				assertNotNull(prereq);
 			}
-			return;
-		}
-
-		if (prereq == null)
-		{
-			fail("Could not parse prereq: '" + aPreString + "'");
-		}
 
-		StringWriter sw = new StringWriter();
-		PrerequisiteWriter writer = new PrerequisiteWriter();
-		try
-		{
+			StringWriter sw = new StringWriter();
+			PrerequisiteWriter writer = new PrerequisiteWriter();
 			writer.write(sw, prereq);
```

Comment:
We still have a problem.  The return statement that used to be on line 492 used to abort when an exception was expected.  I believe after line 433 assertThrows you may need a continue; or some such to ensure that the same logic is followed.  The main problem is prereq can still be null here, and thus we can still try to write a null prereq.